src/main/java/org/broadinstitute/consent/http/models/User.java

```diff
@@ -306,11 +309,11 @@ public void addProperty(UserProperty userProp) {
   }
 
   public void addLibraryCard(LibraryCard card) {
-    if (Objects.isNull(this.getLibraryCards())) {
-      this.setLibraryCards(new ArrayList<>());
+    if (this.libraryCards == null) {
+      this.libraryCards = new ArrayList<>();
```

Comment:
Not sure how consistent the consent code is, but I think using `this.` when it's not needed for disambiguation purposes makes the code harder to read
```suggestion
    if (libraryCards == null) {
     libraryCards = new ArrayList<>();
```

Reply:
The proposed change makes this use inconsistent with the pattern used throughout this class.  My opinion is that I agree with your assessment that it's easier to read the code without "this."  The open question is do I trade off the time to update the entire class, make the single change, or leave it as-is?

Reply:
The other thing in the back of my mind (and why I'm holding off on your other note about a test refactor) is that I'm wondering if the builder pattern would be better for some of these classes.  I'm still working through that in a mental exercise.

Reply:
> I'm wondering if the builder pattern would be better for some of these classes

My general philosophy is that all data classes like this should be immutable. If you need to mutate, you create a copy. Adding builders is one way to support immutable objects, it depends on the needs of the code. Often I find that the only need for builders is for tests, in which case moving the builder-type code into test classes is a good way to make that clear.